wallets/index.js

```diff
@@ -106,46 +110,120 @@ export function useWallet (name) {
+}
+
+function extractClientConfig (fields, config) {
+  return extractConfig(fields, config, true)
+}
+
+function extractServerConfig (fields, config) {
+  return extractConfig(fields, config, false)
+}
+
 function useConfig (wallet) {
   const me = useMe()
 
   const storageKey = getStorageKey(wallet?.name, me)
-  const [localConfig, setLocalConfig, clearLocalConfig] = useLocalConfig(storageKey)
+  const [clientConfig, setClientConfig, clearClientConfig] = useClientConfig(storageKey)
 
   const [serverConfig, setServerConfig, clearServerConfig] = useServerConfig(wallet)
 
-  const hasLocalConfig = !!wallet?.sendPayment
+  const hasClientConfig = !!wallet?.sendPayment
   const hasServerConfig = !!wallet?.walletType
 
-  const config = {
-    // only include config if it makes sense for this wallet
-    // since server config always returns default values for autowithdraw settings
-    // which might be confusing to have for wallets that don't support autowithdraw
-    ...(hasLocalConfig ? localConfig : {}),
-    ...(hasServerConfig ? serverConfig : {})
+  let config = {}
+  if (hasClientConfig) config = clientConfig
+  if (hasServerConfig) {
+    const { enabled } = config || {}
+    config = {
+      ...config,
+      ...serverConfig
+    }
+    // wallet is enabled if enabled is set in client or server config
+    config.enabled ||= enabled
   }
 
-  const saveConfig = useCallback(async (config) => {
-    if (hasLocalConfig) setLocalConfig(config)
-    if (hasServerConfig) await setServerConfig(config)
-  }, [wallet])
+  const saveConfig = useCallback(async (newConfig, { logger }) => {
+    // NOTE:
+    //   verifying the client/server configuration before saving it
+    //   prevents unsetting just one configuration if both are set.
+    //   This means there is no way of unsetting just one configuration
+    //   since 'detach' detaches both.
+    //   Not optimal UX but the trade-off is saving invalid configurations
+    //   and maybe it's not that big of an issue.
+    if (hasClientConfig) {
+      const newClientConfig = extractClientConfig(wallet.fields, newConfig)
+
+      let valid = true
+      try {
+        await walletValidate(wallet, newClientConfig)
+      } catch {
+        valid = false
+      }
 
-  const clearConfig = useCallback(async () => {
-    if (hasLocalConfig) clearLocalConfig()
+      if (valid) {
+        setClientConfig(newClientConfig)
+        logger.ok(wallet.isConfigured ? 'payment details updated' : 'wallet attached for payments')
+        if (newConfig.enabled) wallet.enablePayments()
+        else wallet.disablePayments()
+      }
+    }
+    if (hasServerConfig) {
+      const newServerConfig = extractServerConfig(wallet.fields, newConfig)
+
+      let valid = true
+      try {
+        await walletValidate(wallet, newServerConfig)
+      } catch {
+        valid = false
+      }
+
+      if (valid) await setServerConfig(newServerConfig)
+    }
+  }, [setClientConfig, setServerConfig, wallet])
```

Comment:
This is missing `has...Config` deps.

Reply:
Also, couldn't this set or unset local/server config if the fields are blank? Or am I missing something?

Reply:
> This is missing `has...Config` deps.

Done in 32c9f8c0.

> Also, couldn't this set or unset local/server config if the fields are blank? Or am I missing something?

If the fields are blank, `valid` will be false and `setLocal/ServerConfig` will not be called.

I also wanted to avoid saving invalid configurations so we can continue to assume that configurations are valid if they were saved.